HCCGo/app/js/fileManageService.js

```diff
@@ -230,13 +232,40 @@ fileManageService.factory('fileManageService',['$log', '$q', '$routeParams', 'co
               service.diskAvail = Math.floor((info.free/info.total)*100);
           });
       });
-   }
+   };
    
    service.verifyDownloadCancel = function () {
 	  service.userDownAuth = false;
       notifierService.warning('Action cancelled by user.');
-   }
+   };
    
+   service.viewFile = function() {
+        connectionService.getFileSize(service.focus.location).then(function(size){
+            if (size > 5*105*1024) {
+                notifierService.warning("File must be downloaded to be viewed.", "File too big to view!");
+            }
+            else {
+                let file = "/" + service.focus.location.substring(service.focus.location.lastIndexOf("/"), service.focus.location.length);
+                tmp.dir({prefix: 'hcc_tmp', unsafeCleanup: true}, function _tempDirCreated(err, path, cleanupCallback) {
+                    connectionService.quickDownload(service.focus.location, path + file).then(function(flag) {
+                        if (flag) {
+                            shell.openItem(path + file);
+                        }
+                        else {
+                            notifierService.error("Error viewing the file!", "File View Failed!");
+                        }
+                        $timeout(() => {
+                            shell.moveItemToTrash(path);
+                        }, 3000);
+                    });
+                });
+            }
+        });
+    };
+
+   service.isDirectory = function(id) {
+       angular.element("#btnView").attr('disabled', '');
```

Comment:
I think this should be removed (along with the other css attr changes).  Instead, ng-disabled in the HTML should be used.  I'm not sure what element in the `focusFile` object would say it's a directory.